ko gives a very obscure error. If you install ko with the wrong golang version.

## What version of Go are you using (go version)?
```bash
$ go version
go version go1.10.4 linux/amd64
```
Does this issue reproduce with the latest release?
Yes.

What operating system and processor architecture are you using (go env)?

```bash
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/vagrant/.cache/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/vagrant/go"
GORACE=""
GOROOT="/usr/lib/go-1.10"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go-1.10/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build100699947=/tmp/go-build -gno-record-gcc-switches"
```
What did you do?
I installed a vm with ubuntu and used `apt-get install golang-go`

What did you expect to see?
I expected `GO111MODULE=on go get github.com/google/ko/cmd/ko` to install ko

What did you see instead?

```bash
# github.com/google/ko/vendor/github.com/google/go-containerregistry/pkg/logs
go/src/github.com/google/ko/vendor/github.com/google/go-containerregistry/pkg/logs/logs.go:38:10: l.Writer undefined (type *log.Logger has no field or method Writer)
# github.com/google/ko/vendor/github.com/docker/cli/cli/config
go/src/github.com/google/ko/vendor/github.com/docker/cli/cli/config/config.go:109:18: undefined: os.UserHomeDir
# github.com/google/ko/vendor/k8s.io/apimachinery/pkg/util/errors
go/src/github.com/google/ko/vendor/k8s.io/apimachinery/pkg/util/errors/errors.go:99:10: undefined: errors.Is
# github.com/google/ko/vendor/sigs.k8s.io/structured-merge-diff/v3/value
go/src/github.com/google/ko/vendor/sigs.k8s.io/structured-merge-diff/v3/value/mapreflect.go:95:13: val.MapRange undefined (type reflect.Value has no field or method MapRange)
go/src/github.com/google/ko/vendor/sigs.k8s.io/structured-merge-diff/v3/value/mapreflect.go:173:14: rhs.MapRange undefined (type reflect.Value has no field or method MapRange)
go/src/github.com/google/ko/vendor/sigs.k8s.io/structured-merge-diff/v3/value/mapreflect.go:194:13: lhs.MapRange undefined (type reflect.Value has no field or method MapRange)
```

## Fix
Install golang version 1.15 and ko will install properly. This error does not make this fix obvious. I also do not see the golang version requirements in the documentation.

--- a/README.md
+++ b/README.md
@@ -12,6 +12,8 @@
 
 `ko` can be installed and upgraded by running:
 
+**Note**: Golang version `1.12.0` or higher is required.
+
 ```shell
 GO111MODULE=on go get github.com/google/ko/cmd/ko
 ```

